torch/_subclasses/fake_utils.py

```diff
@@ -75,6 +76,74 @@ def is_sdpa_error(func, idx, e):
     return False
 
 
+def try_convert_fake_to_real(
+    ten_list: List[Union[FakeTensor, Any]]
+) -> List[Union[FakeTensor, torch.Tensor, Any]]:
+    """
+    Attempt to convert fake tensors to a corresponding real tensor with the correct underlying storage by looking up
```

Comment:
Can you explain why you have to do something this complicated? A trivial alternate implementation would be to have fake tensor maintain weakref to the real tensor they were converted from and read it off from that. Why do we have to do something so fancy?

Reply:
Aot autograd will do weird things with views and such. this is more reliable.